Libraries/SparkFun_ATECCX08a_Arduino_Library/src/SparkFun_ATECCX08a_Arduino_Library.cpp

```diff
@@ -0,0 +1,1201 @@
+  uint8_t data3[] = {0x30, 0x00, 0x30, 0x00}; 
+  result3 = write(ZONE_CONFIG, (116 / 4), data3, 4);
+
+  // set slot config on slot 10 and 11 
+  //slotconfig for slot 10: is secret, not encryptread, no usage limitation, can be used by all commands, write config never
+  //slotconfig for slot 11: is secret, not encryptread, no usage limitation, can be used by all commands, write config never
+  uint8_t data4[] = {0x00, 0x20, 0x00, 0x20}; 
+  result4 = write(ZONE_CONFIG, (40 / 4), data4, 4);
+
+
+  return (result1 && result2 && result3 && result4);
+}
+
+
+boolean ATECCX08A::loadPublicKey(uint8_t *data, bool debug)
+{
+	uint8_t public_x[32];
+	uint8_t public_y[32];
+	for(int i =0; i <32; i++)
+	{
+		public_x[i]=data[i];
+		public_y[i]=data[i+32];
+	}
+
+	//Now Write first 32 bytes of public key to slot 10
+	sendCommand(COMMAND_OPCODE_WRITE, WRITE_DATA_32, ADDRESS_DATA_READ_SLOT10_BLOCK_0, public_x, 32);
+	delay(100);
+	if(receiveResponseData(4) == false) return false;
+	idleMode();
+	if(checkCount() == false) return false;
+	if(checkCrc() == false) return false;
+	if(inputBuffer[1] == 0x00) {
+	if (debug) Serial.println("Loaded Public Key X Component Successfully");
+	//return true;   // If we hear a "0x00", that means it had a successful write
+	}
+
+	//Now Write second 32 bytes of public key to slot 10
+	sendCommand(COMMAND_OPCODE_WRITE, WRITE_DATA_32, ADDRESS_DATA_READ_SLOT10_BLOCK_1, public_y, 32);
+	delay(100);
+	if(receiveResponseData(4) == false) return false;
+	idleMode();
+	if(checkCount() == false) return false;
+	if(checkCrc() == false) return false;
+	if(inputBuffer[1] == 0x00) {
+	if (debug) Serial.println("Loaded Public Key Y Component Successfully");
+	return true;   // If we hear a "0x00", that means it had a successful write
+	}
+	else return false;
+}
+
+boolean ATECCX08A::readPublicKey(boolean debug)
+{
+  // read block 0, the first 32 bytes of slot10 into inputBuffer
+  read(ZONE_DATA, ADDRESS_DATA_READ_SLOT10_BLOCK_0, 32); 
+  // copy current contents of inputBuffer into storedPublicKey[] (for later viewing/comparing)
+  memcpy(&storedPublicKey[0], &inputBuffer[1], 32);
+
+  // read block 0, the first 32 bytes of slot10 into inputBuffer
+  read(ZONE_DATA, ADDRESS_DATA_READ_SLOT10_BLOCK_1, 32); 
+  // copy current contents of inputBuffer into storedPublicKey[] (for later viewing/comparing)
+  memcpy(&storedPublicKey[32], &inputBuffer[1], 32);
+
+  if(debug)
+  {
+    Serial.println("storedPublicKey: ");
+    for (int i = 0; i < sizeof(storedPublicKey) ; i++)
+    {
+      Serial.print(i);
+    Serial.print(": 0x");
+    if((storedPublicKey[i] >> 4) == 0) Serial.print("0"); // print preceeding high nibble if it's zero
+    Serial.print(storedPublicKey[i], HEX); 
+    Serial.print(" \t0b");
+    for(int bit = 7; bit >= 0; bit--) Serial.print(bitRead(storedPublicKey[i],bit)); // print binary WITH preceding '0' bits
+    Serial.println();
+    }
+    Serial.println();
+  }
+}
```

Comment:
at end of this function we need
"return true;"
to avoid compiler warnings

I have an idea for bitwise anding results to verify successful reads, will ask Dr. Daily about it